content/FIDO2/FIDO2_WebAuthn_Developer_Guide/WebAuthn_Client_Registration.adoc

```diff
@@ -0,0 +1,188 @@
+
+**challenge:** In order to prevent replay attacks, the value should be randomly generated by the RP and encoded in the base64 url safe format.
+
+**excludeCredentials:** limits creation of multiple credentials for the same account on a single authenticator. If the authenticator finds an existing credential **type** and **id** for the RP, then a new credential will not be created.
+
+**pubKeyCredParams:** The RP can specify the desired properties of the credential to be created. **type:** only one type: “public-key”. **alg:** cryptographic signature algorithm preference specified by https://www.iana.org/assignments/cose/cose.xhtml#algorithms[COSE Algorithms registry].
+
+**rp:** relying party information. **name** is required. The **id** attribute must be the origin seen by the client, or a subdomain of it. If id is omitted, then origins effective domain is used.
+
+**timeout:** the time, in milliseconds, that the caller is willing to wait for the call to complete.
+
+**user:** contains data about the user account. The **displayName** provides a user-friendly name which may be displayed to the user. The **name** attribute takes a traditional username or the like, to disambiguate accounts with similar **displayNames**.The **id** attribute is the userHandle that is generated by the RP and used to uniquely identify a user. Unlike **name**, **id** is not meant to be human-readable and should not contain personally identifying information such as an e-mail address.
+
+**2.** The JavaScript client calls navigator.credentials.create(). The browser validates the rpId against the origin, hashes the clientData, and calls authenticatorMakeCredential method.
+
+**3.** Before proceeding, the authenticator will ask for some form of user consent. After verifying consent, the authenticator will create a new asymmetric key pair and safely store the private key. The public key becomes part of the attestation, which the authenticator signs over with the private key. The manufacturer’s attestation certificate chain may also be returned so that the relying party can validate the device back to a root of trust.
+
+**4.** The new public key, a credential id, and other attestation data are returned to the browser where they become the attestationObject. 
+
+**5.** The navigator.credentials.create() Promise resolves to a PublicKeyCredential which returned to the RP to finalize the registration.
+
+==== Example
+
+**PublicKeyCredential**
+[source,javascript]
+----
+{
+  "id": "X9FrwMfmzj...",
```

Comment:
Good to know. A added note above the examples that parsed object has binary parameters that are represented as base64 for readability in eacd866. Since the fields are a mix of types, I want to find a better way to convey this to the reader. I will think about it and we can improve it in the next iteration. How does that sound?

Reply:
Sounds good!